back/src/domains/establishment/adapters/PgEstablishmentAggregateRepository.sql.ts

```diff
@@ -126,3 +129,56 @@ LEFT JOIN establishments_contacts AS ec ON uec.uuid = ec.uuid
 LEFT JOIN establishment_locations_agg AS ela ON e.siret = ela.establishment_siret
 ORDER BY e.siret
 `;
+
+export const updateAllEstablishmentScoresQuery = async (
+  db: KyselyDb,
+): Promise<void> => {
+  const minimumScore = 10;
+  const conventionCountCoefficient = 10;
+  const discussionCountCoefficient = 100;
+
+  await db
+    .with("convention_counts", (qb) =>
+      qb
+        .selectFrom("conventions")
+        .where("date_submission", ">=", sql<Date>`NOW() - INTERVAL '1 year'`)
+        .where("status", "=", "ACCEPTED_BY_VALIDATOR")
+        .groupBy("siret")
+        .select([
+          "siret",
+          ({ fn }) => fn.count("siret").as("convention_count"),
+        ]),
+    )
+    .with("discussion_counts", (qb) =>
+      qb
+        .selectFrom("discussions as d")
+        .innerJoin("exchanges", "d.id", "exchanges.discussion_id")
+        .where("d.created_at", ">=", sql<Date>`NOW() - INTERVAL '1 year'`)
+        .select([
+          "siret",
+          sql`COUNT(DISTINCT d.id)`.as("total_discussions"),
+          sql`COUNT(DISTINCT CASE WHEN exchanges.sender = 'establishment'::exchange_role THEN d.id END)`.as(
+            "answered_discussions",
+          ),
+        ])
+        .groupBy("siret"),
+    )
+    .updateTable("establishments as e")
+    .set({
+      score: sql`ROUND(
+          ${minimumScore}
+          + COALESCE((SELECT convention_count * ${conventionCountCoefficient} FROM convention_counts WHERE siret = e.siret), 0)
+          + COALESCE((
+            SELECT
+              CASE
+                WHEN total_discussions > 0
+                THEN (answered_discussions::float / total_discussions) * ${discussionCountCoefficient}
```

Comment:
`answered_discussions` peut être null. Avec postgres, la division null/total_discussions sera null, et une addition avec un null sera null aussi.

je propose:

```suggestion
                THEN (COALESCE(answered_discussions::float, 0) / total_discussions) * ${discussionCountCoefficient}
```

ou sinon de modifier plus haut dans le WITH de discussion_counts:

```sql
sql`COUNT(DISTINCT 
  CASE WHEN exchanges.sender = 'establishment'::exchange_role THEN d.id
  ELSE 0 // par défaut sans ce else, ce sera NULL
  END)`.as(
            "answered_discussions",
          ),
```

Reply:
Il y a bien un `coalesce` mais à la sortie du select. Pour moi ça revient au même. 

J'ai joué la query sur le replicat, et j'ai pas eu de problème (sachant que `score` est obligatoire, on aurait vu un problème en essayant de le mettre à null)